### PR TITLE
HP-38 Make email case-insensitive (and add names to User model)

### PR DESCRIPTION
**Pull request by vkurup**
Commit 1 uses the PG specific `citext` field to store the email address case-insensitively, and adds a test that broke before the change and passes after it.

Commit 2 adds first_name and last_name back to the user model. They were removed in the jade-truffle templated files, but since they are present in the Wagtail admin, seems like we should use them.

### CODE REVIEW

**Review by dchukhin — COMMENTED** [review]
Looks good, though I'll request some minor method/test updates

File: apps/users/models.py [hunk 1]
```diff
@@ -7,7 +8,11 @@
 
 
 class User(AbstractBaseUser, PermissionsMixin, IndexedTimeStampedModel):
-    email = models.EmailField(max_length=255, unique=True)
+    # use special Postgres-only Case Insensitive Email Field
+    # https://docs.djangoproject.com/en/dev/ref/contrib/postgres/fields/#django.contrib.postgres.fields.CIEmailField
+    email = CIEmailField(max_length=255, unique=True)
+    first_name = models.CharField(_("first name"), max_length=150, blank=True)
```
[the remark this answers is not in the document]
Comment (vkurup) [reply apps/users/models.py:14]: Good thinking!

**Comment by vkurup** [comment]
@dchukhin I made updates based on your review in the first new commit (using the exact code from [Django's versions of these methods](https://github.com/django/django/blob/master/django/contrib/auth/models.py#L373-L382).

The next 3 commits are my first attempt to use pytest fixtures in combination with Factory Boy. I haven't used pytest much before, so I don't know if this is the best way to do this. Happy to make adjustments, or we can discuss how to do this for future PRs.

**Review by dchukhin — APPROVED** [review]
✉️

**Comment by dchukhin** [comment]
Looks good to me. I don't really know the best way to use fixtures for this project at the moment, but happy to talk about it for future pull requests

**Pull request merged**